server/cluster/hot_region_storage_test.go

```diff
@@ -0,0 +1,395 @@
+	}
+	if err != nil {
+		b.Fatal(err)
+	}
+	b.StopTimer()
+}
+
+func BenchmarkCompaction(b *testing.B) {
+	// delete data in between today and tomrrow
+	regionStorage, clear, err := newTestHotRegionStorage(10*time.Hour, 30)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer clear()
+	raft := regionStorage.cluster
+	endTime := time.Now()
+	regions := newTestHotRegions(1000, 3)
+	for _, region := range regions {
+		raft.putRegion(region)
+	}
+	// leveldb will compaction after 30 times delete
+	for i := 0; i < defaultCompactionTime-1; i++ {
+		// 144=24*60/10
+		endTime = writeIntoDB(regionStorage, regions, 144, endTime)
+		regionStorage.delete()
+		regionStorage.remianedDays--
+	}
+	b.ResetTimer()
+	regionStorage.delete()
+	b.StopTimer()
+}
+
+func BenchmarkTwoTimesCompaction(b *testing.B) {
+	// delete data in between today and tomrrow
+	regionStorage, clear, err := newTestHotRegionStorage(10*time.Hour, 30)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer clear()
+	raft := regionStorage.cluster
+	endTime := time.Now()
+	//4464=(60*24*31)/10
+	regions := newTestHotRegions(1000, 3)
+	for _, region := range regions {
+		raft.putRegion(region)
+	}
+	//leveldb will compaction after 30 times delete
+	for i := 0; i < 2*defaultCompactionTime-1; i++ {
+		//144=24*60/10
+		endTime = writeIntoDB(regionStorage, regions, 144, endTime)
+		regionStorage.delete()
+		regionStorage.remianedDays--
+	}
+	b.ResetTimer()
+	regionStorage.delete()
+	b.StopTimer()
+}
+
+func BenchmarkDeleteAfterYear(b *testing.B) {
+	regionStorage, clear, err := newTestHotRegionStorage(10*time.Hour, -1)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer clear()
+	raft := regionStorage.cluster
+	regions := newTestHotRegions(1000, 3)
+	for _, region := range regions {
+		raft.putRegion(region)
+	}
+	endTime := time.Now()
+	//4464=(60*24*31)/10
+	endTime = writeIntoDB(regionStorage, regions, 4464, endTime)
+	//334=365-31
+	for i := 0; i < 334; i++ {
+		regionStorage.delete()
+		//144=24*60/10
+		endTime = writeIntoDB(regionStorage, regions, 144, endTime)
+	}
+	b.ResetTimer()
+	regionStorage.delete()
+	b.StopTimer()
+}
+
+func newTestHotRegionStorage(pullInterval time.Duration, remianedDays int64) (
+	hotRegionStorage *HotRegionStorage,
+	clear func(), err error) {
+	writePath := "./tmp"
+	ctx := context.Background()
+	_, opt, err := newTestScheduleConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+	raft := newTestCluster(ctx, opt).RaftCluster
+	//delete data in between today and tomrrow
+	hotRegionStorage, err = NewHotRegionsStorage(ctx,
+		writePath, nil, raft, nil, remianedDays, pullInterval)
+	if err != nil {
+		return nil, nil, err
+	}
+	clear = func() {
+		hotRegionStorage.Close()
+		PrintDirSize(writePath)
+		os.RemoveAll(writePath)
+	}
+	return
+}
+func writeIntoDB(regionStorage *HotRegionStorage,
+	regions []*core.RegionInfo, times int,
+	endTime time.Time) time.Time {
+	raft := regionStorage.cluster
+	for i := 0; i < times; i++ {
+		if i%1000 == 0 {
+			fmt.Println(i)
```

Comment:
use log?

Reply:
This is the code that has not been deleted before. It has been deleted